config/src/network_id.rs

```diff
@@ -189,27 +189,17 @@ impl NetworkId {
     }
 
     /// Roles for P2P relationships
-    pub fn p2p_roles(&self, role: &RoleType) -> &'static [PeerRole] {
+    pub fn p2p_roles(&self) -> &'static [PeerRole] {
         match self {
             NetworkId::Validator => &[PeerRole::Validator],
```

Comment:
if this is true, it seems much easier to just contain the check that on the Validator network that they are both validators and peers, no? no point for this function

Reply:
I added this function because you were very adamant that I couldn't put `PeerRole::Validator` as an `Upstream` role.  This added extra complexity where both in the upstream role function I had to look up the network, and say nothing.  Then *everywhere* it was used put an if statement, checking if it was a validator talking to another validator.

If we're now fine saying that a validator is both an upstream and a downstream of a validator directly, I can remove this function.

Reply:
Removed.